### DISCUSSION TITLE
Dynamically set input files with Detekt Gradle Plugin

### DISCUSSION THREAD

**Discussion by PhilippNowak96** [detekt/detekt#5454] [Q&A] [answered]
Is there a way to dynamically set input files (like --input param for CLI) but with Gradle plugin? My goal is to only scan files which have been changed on a specific branch, so I am using some `git diff` stuff to create a list of changed file paths and now I need to feed it into Detekt Gradle plugin.

**Comment by PhilippNowak96** [comment, marked as answer]
I just found out how to apply custom plugins for CLI and configure custom reports. That was my initial reason to use the gradle plugin. Now that I can use all the necessary things with CLI, I can simply use the `--input` param. So this question is kinda obsolete.

**Comment by TWiStErRob** [comment]
A word of warning: if you're running Detekt with type resolution, the changed files don't cover the files that could trigger new Detekt findings. e.g.

file1.kt
```kotlin
fun f(): List<String> = TODO("implementation omitted for brevity")
```
file2.kt
```kotlin
class c {
    var list = f()
}
```

PR changes file1.kt:
```diff
-fun f(): List<String> = TODO("implementation omitted for brevity")
+fun f(): MutableList<String> = TODO("implementation omitted for brevity")
```

`var list = f()` **in file2.kt** would be flagged with [DoubleMutabilityForCollection](https://detekt.dev/docs/next/rules/potential-bugs#doublemutabilityforcollection).
